extensions/amp-form/0.1/amp-form.js

```diff
@@ -268,21 +267,24 @@ export class AmpForm {
       this.validator_.onBlur(e);
     }, true);
 
-    const afterVerifierCommit = () => {
-      // Move from the VERIFYING state back to INITIAL
-      if (this.state_ === FormState_.VERIFYING) {
-        this.setState_(FormState_.INITIAL);
-      }
-    };
     this.form_.addEventListener('change', e => {
-      this.verifier_.onCommit()
-          .then(updatedElements => {
-            updatedElements.forEach(checkUserValidityAfterInteraction_);
-            this.validator_.onBlur(e);
-          }, () => {
-            checkUserValidityAfterInteraction_(dev().assertElement(e.target));
-          })
-          .then(afterVerifierCommit, afterVerifierCommit);
+      this.verifier_.onCommit().then(({updatedElements, errors}) => {
+        // Only make the verify XHR if the user hasn't pressed submit.
+        if (this.state_ === FormState.VERIFYING) {
+          if (errors.length) {
+            this.setState_(FormState.VERIFY_ERROR);
+            this.renderTemplate_(
+                /** @type {!JsonObject} */ ({verifyErrors: errors}));
+            this.triggerAction_(FormEvent.VERIFY_ERROR, errors);
+          } else {
+            this.setState_(FormState.INITIAL);
```

Comment:
Any consequence of changing this order of operations?

Reply:
Oops you're right the state should be changed after all the elements have been updated.